nemo/collections/multimodal/speech_llm/data/lhotse_dataset.py

```diff
@@ -79,102 +81,108 @@ def __init__(
+                    "audio_signal": audio,
+                    "audio_signal_length": audio_lens,
+                    "audio_ratio": torch.FloatTensor(audio_ratio),
+                    "metadata": metadata,
+                    **collated_text_data,
+                }
+            )
+            ans.update(return_batch)
+
+        # convert text examples to tensors
+        text_examples = all_cuts.filter(lambda c: isinstance(c, (SourceTargetTextExample, NeMoSFTExample)))
+        if text_examples:
+            pad_id = self.text_processor.pad_id
+            text_minibatch = dict(
+                text_input_ids=collate_vectors_lhotse([e.input_ids for e in text_examples], padding_value=pad_id),
+                text_input_lens=torch.tensor([len(e.input_ids) for e in text_examples], dtype=torch.int64),
+                text_answer_ids=collate_vectors_lhotse([e.answer_ids for e in text_examples], padding_value=pad_id),
+                text_answer_lens=torch.tensor([len(e.answer_ids) for e in text_examples], dtype=torch.int64),
+                text_context_ids=collate_vectors_lhotse([e.context_ids for e in text_examples], padding_value=pad_id),
+                text_context_lens=torch.tensor([len(e.context_ids) for e in text_examples], dtype=torch.int64),
+                text_masks=collate_vectors_lhotse([e.mask for e in text_examples], padding_value=0),
+            )
+            ans.update(text_minibatch)
+
+        return ans
 
 
 def collate_text_data(
     cuts,
     default_context: str,
-    text_processor: TextProcessing,
+    text_processor: PromptFormatterTextProcessing,
     tokens_to_generate: int,
     pad_to_max_length: bool,
     max_seq_length: int,
 ) -> dict:
     """Perform text collation equivalent to nemo/collections/multimodal/data/audio_text_qa_dataset.py:121"""
     batch_size = len(cuts)
     pad_id = text_processor.pad_id
-    examples = [
-        {
-            k: torch.as_tensor(v)
-            for k, v in text_processor._process_example(
-                context=cut.context,
-                output=cut.supervisions[0].text,
-            ).items()
-        }
-        for cut in cuts
-    ]
+    examples = [{k: torch.as_tensor(v) for k, v in text_processor._process_example(cut).items()} for cut in cuts]
     fields = as_dict(examples)
 
     def get_max_len(input_list):
         return max([len(x) for x in input_list])
 
-    max_length = tokens_to_generate + max(
-        get_max_len(fields["input_ids"]), get_max_len(fields["context_ids"]), get_max_len(fields["answer_ids"])
-    )
-    # increase max length to nearest multiple of 4 or 8
+    input_id_maxlen = get_max_len(fields["input_ids"])
+    context_id_maxlen = tokens_to_generate + get_max_len(fields["context_ids"])
+    answer_id_maxlen = get_max_len(fields["answer_ids"])
     if pad_to_max_length:
-        max_length = max_seq_length
-    else:
-        max_length = min(max_seq_length, ceil_to_nearest(max_length, 8))
```

Comment:
the `ceil_to_nearest()` is needed of sequence parallel, although we currently don't support sequence parallel in speechlm, we might need to add this back in the future

Reply:
Let's restore it when adding sequence parallel support